src/nvim/charset.c

```diff
@@ -734,8 +735,8 @@ int vim_strnsize(char_u *s, int len)
 /// @return Number of characters.
 #define RET_WIN_BUF_CHARTABSIZE(wp, buf, p, col) \
   if (*(p) == TAB && (!(wp)->w_p_list || lcs_tab1)) { \
-    int ts; \
-    ts = (buf)->b_p_ts; \
+    int64_t ts; \
+    ts = (int)(buf)->b_p_ts; \
```

Comment:
It is better to merge this with the previous line. I also see no reason to have `int64_t` here since you cast it to `int` below (and, BTW, with `int64_t ts` it is not correct to cast to `int` on this line): I would suggest to simply write this as

    const int ts = (int)  (buf)->b_p_ts;
    return (ts - (((int) col) % ts));

(reasoning: return type is int). Or as

    const colnr_T ts = (colnr_T) (buf)->b_p_ts;
    return (int) (ts - (col % ts));

(reasoning: this thing is operating on columns, but it is not guaranteed that colnr_T == int).